Add documentation about changing cluster node hostname

ycointe opened the pull request:
Issue: GH-1162

**Component**:

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: Add documentation

**Summary**: How to change a node hostname in a cluster

**Acceptance criteria**: 


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #1162

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

ycointe committed:
Add documentation about changing cluster node hostname

Issue: GH-1162

bert-e commented:
# Hello ycointe,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1162-doc-change-name-node` into
`development/2.0`
* [w/2.1/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1162-doc-change-name-node) will be merged into `development/2.1` 
* [w/2.2/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1162-doc-change-name-node) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @alexandre-allard-scality

* @gdemonet

* @zempashi

* @sayf-eddine-scality

* @slaperche-scality

* @alxf

* @monpote

* @tcarmet

* @carlito-scality

* @ramanduh

* @cmonfort

* @thomasdanan

* @lucieleonard

* @nicolast

* @teddyandrieux

* @nootal

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/1162-doc-change-name-node` has diverged from
`development/2.0` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.0` into `improvement/1162-doc-change-name-node`
* Rebase `improvement/1162-doc-change-name-node` onto `origin/development/2.0`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+Change the hostname of a cluster node
+=====================================
+
+On the node
```
`s/on/from` ?

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+           Architecture: x86-64
+
+
+On the bootstrap node
```
`s/on/from` ?

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+.. code-block:: shell
+
+   $ kubectl get nodes
+   node2.novalocal    NotReady   etcd,master                   19h       v1.11.7
```
Missing the header here (i.e. `NAME  STATUS  ROLES  AGE  VERSION`)

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+     name: <NEW HOSTNAME>
+     annotations:
+       metalk8s.scality.com/ssh-key-path: /etc/metalk8s/pki/id_rsa
+       metalk8s.scality.com/ssh-host: <IP Address>
```
IP shouldnt be a variable, so let's pick a random one

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+   $ kubectl get nodes
+   node2.novalocal    NotReady   etcd,master                   19h       v1.11.7
+
+Change the name of the node in the YAML file used to created it
```
```suggestion
Change the name of the node in its manifest file.
```

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+
+   $ kubectl apply -f new-node-master3.yaml
+
+Delete the old node (here ``node2.novalocal``):
```
```suggestion
Delete the previous definition of the node (here ``node2.novalocal``) in Kubernetes:
```

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+
+   $ kubectl delete node node2.novalocal
+
+Log on the salt master:
```
```suggestion
Obtain a shell into the Salt master container:
```

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+
+   $ kubectl -it -n kube-system exec salt-master-master.novalocal bash
+
+Then:
```
```suggestion
Execute the node deployment procedure with the node's new name:
```

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+Change the hostname of a cluster node
```
```suggestion
Changing the hostname of a MetalK8s node
```

ycointe replied:
Done

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+
+.. code-block:: shell
+
+        $ hostnamectl set-hostname <your hostname, here: ControlPlane2>
```
A code block's body should be indented by 3 characters only (so the contents line up with the `code-block` directive)

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+
+.. code-block:: shell
+
+        $ hostnamectl status
```
Ditto code block

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+   $ salt-run state.orchestrate metalk8s.orchestrate.deploy_node saltenv=metalk8s-2.0 pillar="{'orchestrate': {'node_name': 'controlplane2'}}"
+
+
+           Summary for master.novalocal_master
```
Ditto code block

jbertran commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,112 @@
+           Total run time:  132.435 s
+
+
+On the Control Plane node delete the certificates:
```
Not sure I understand this as is. Do you mean the following?

> If the node on which the operation is done is a control-plane node, delete the certificates:

gdemonet replied:
No, that's just about re-creating certificates for the changed node:

```suggestion
On the updated node, delete the now invalid certificates:
```

ycointe replied:
I just checked ! the certificate have been to be deleted on the controlPlane (the controlplane you just change the hostname), so not on the bootstrap !

ycointe replied:
(didn't reload my page before to comment... agree with @gdemonet )

jbertran reviewed (changes requested):
Nice work on the procedure!

I didn't test it, so my comments are purely on grammar.

bert-e commented:
# History mismatch



Merge commit #bb6c439cda093cda0bf7baa085f3b595345be9c7 on the integration branch
`w/2.1/improvement/1162-doc-change-name-node` is merging a branch which is neither the current
branch `improvement/1162-doc-change-name-node` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/1162-doc-change-name-node` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

jbertran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1162-doc-change-name-node` into
`development/2.0`
* [w/2.1/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1162-doc-change-name-node) will be merged into `development/2.1` 
* [w/2.2/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1162-doc-change-name-node) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @alexandre-allard-scality

* @gdemonet

* @zempashi

* @sayf-eddine-scality

* @slaperche-scality

* @alxf

* @monpote

* @tcarmet

* @carlito-scality

* @ramanduh

* @cmonfort

* @thomasdanan

* @lucieleonard

* @nicolast

* @teddyandrieux

* @nootal





The following reviewers are expecting changes from the author, or must review again:

* @jbertran

bert-e commented:
# History mismatch



Merge commit #8d4a6df692fe2813f065d5f12a043532bdefbcd4 on the integration branch
`w/2.1/improvement/1162-doc-change-name-node` is merging a branch which is neither the current
branch `improvement/1162-doc-change-name-node` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/1162-doc-change-name-node` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

jbertran reviewed (dismissed):
LGTM

ycointe commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1162-doc-change-name-node` into
`development/2.0`
* [w/2.1/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1162-doc-change-name-node) will be merged into `development/2.1` 
* [w/2.2/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1162-doc-change-name-node) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @alexandre-allard-scality

* @gdemonet

* @zempashi

* @sayf-eddine-scality

* @slaperche-scality

* @alxf

* @monpote

* @tcarmet

* @carlito-scality

* @ramanduh

* @cmonfort

* @thomasdanan

* @lucieleonard

* @nicolast

* @teddyandrieux

* @nootal

ycointe commented:
@bert-e approve

gdemonet commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,115 @@
+   NAME               STATUS     ROLES                         AGE       VERSION
+   node2.novalocal    NotReady   etcd,master                   19h       v1.11.7
+
+Change the name of the node in its manifest file:
```
Would be achievable in a single step with `kubectl edit node node2.novalocal`

gdemonet replied:
Won't work actually, can't edit a node name.

gdemonet commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,115 @@
+   Total run time:  132.435 s
+
+
+On the updated node, delete the now invalid certificates:
```
This needs to be done before re-deployment

gdemonet commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,115 @@
+
+   [bootstrap]$ rm -f /var/lib/kubelet/pki/*.pem
+
+Then restart the kubelet service:
```
This will (or should, at least) be done by Salt when you re-generate the certificates

gdemonet commented on docs/changenodehostname.rst:
```diff
@@ -0,0 +1,115 @@
+
+.. code-block:: shell
+
+   [bootstrap]$ systemctl restart kubelet
```
(not related to this line)

@NicolasT also mentioned (e-mail) that we should remove the previous minion key from the master (before deployment as well would be good)

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @alexandre-allard-scality

* @gdemonet

* @zempashi

* @sayf-eddine-scality

* @slaperche-scality

* @alxf

* @monpote

* @tcarmet

* @carlito-scality

* @ramanduh

* @cmonfort

* @thomasdanan

* @lucieleonard

* @nicolast

* @teddyandrieux

* @nootal





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet







*The following options are set:* **approve**

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/1162-doc-change-name-node` has diverged from
`development/2.0` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.0` into `improvement/1162-doc-change-name-node`
* Rebase `improvement/1162-doc-change-name-node` onto `origin/development/2.0`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.




*The following options are set:* **approve**

lucieleonard commented on docs/index.rst:
```diff
@@ -21,7 +21,7 @@ Welcome to MetalK8s's documentation!
    installation-guide/index
    developer/index
-
+   changenodehostname
```
forgot to change that in my commit. Now named: change_node_hostname.
Even should be renamed: changing_node_hostname

jbertran replied:
@lucieleonard :

Talking about commits: we try to stick to issue # in the body, and the title as the area of intervention + a short description.

We'd rewrite your commit with something along the lines of 

```
docs: rework hostname change procedure

* procedure becomes a list of steps
* [...]

Issue: #1162 
```

Additionally, it would be great to have a separate commit for the rename, so that the diff on the commit that actually makes changes to the file is not just 1 file deleted and 1 added, but rather the actual rework that happened.

lucieleonard commented on docs/change_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+      $ hostnamectl set-hostname <New hostname>
+      $ systemctl restart systemd-hostnamed
+
+#. Check that the hostname has changed
```
Check that the change is taken into account.

lucieleonard commented on docs/change_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+         Architecture: x86-64
+
+
+#. On the bootstrap node, verify the hostname change allowed a change of status on the bootstrap,
```
`#. On the bootstrap node, check the hostname edition allowed a change of status on the bootstrap. The node must be in a **NotReady** status.`

lucieleonard commented on docs/change_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+      $ kubectl get nodes
+      node2    NotReady   etcd,master                   19h       v1.11.7
+
+#. Change the name of the node in the YAML file used to created it
```
`Change the name of the node in the ``yaml`` file used to create it.`

lucieleonard commented on docs/change_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+
+      $ kubectl -it -n kube-system exec salt-master-bootstrap bash
+
+#. Then delete the old node key:
```
remove "then"

lucieleonard commented on docs/change_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+
+      $ salt-key -d <node_key_name>
+
+#. Then run the new node name deployment: 
```
Remove "then"

ycointe committed:
Organize the procedure as a list

Changes done by Lucie

ycointe committed:
Add changing_node_hostname to the operation documentation

Add the documentation, change the different indexes.

bert-e commented:
# History mismatch



Merge commit #1a56dc90fd7c5dbfd988152ad62ca977526b12b8 on the integration branch
`w/2.1/improvement/1162-doc-change-name-node` is merging a branch which is neither the current
branch `improvement/1162-doc-change-name-node` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/1162-doc-change-name-node` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

jbertran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1162-doc-change-name-node` into
`development/2.0`
* [w/2.1/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1162-doc-change-name-node) will be merged into `development/2.1` 
* [w/2.2/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1162-doc-change-name-node) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @nootal

* @cmonfort

* @lucieleonard

* @alxf

* @carlito-scality

* @alexandre-allard-scality

* @teddyandrieux

* @slaperche-scality

* @thomasdanan

* @zempashi

* @nicolast

* @sayf-eddine-scality

* @monpote

* @gdemonet

* @tcarmet





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet







*The following options are set:* **approve**

ycointe commented:
@bert-e approve

ycointe commented:
@bert-e approve

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+
+#. Run the new node name deployment:
+
+   .. code-block:: shell
```
You need a `.. parsed-literal::` directive to have the substition on `|release|`

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+
+#. Change the name of the node in the ``yaml`` file used to create it.
+
+   .. code-block:: yaml
```
ditto, `parsed-literal` is needed here

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+
+      $ hostnamectl status
+
+      Static hostname: controlplane2
```
Make sure to use the same placeholders across the document, to help users see what should be verified.

```suggestion
      Static hostname: <new_hostname>
```

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+      $ hostnamectl status
+
+      Static hostname: controlplane2
+      Pretty hostname: ControlPlane2
```
Where does this come from? Is it automatically set? Did you choose it when running `set-hostname`?

ycointe replied:
yes, I replaced it by <New hostname> as requested in the previous comment

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+         Architecture: x86-64
+
+
+#. On the bootstrap node, check the hostname edition allowed a change of status
```
```suggestion
#. On the bootstrap node, check the hostname edition incurred a change of status.
```

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+
+
+#. On the bootstrap node, check the hostname edition allowed a change of status
+   on the bootstrap. The node must be in a **NotReady** status.
```
```suggestion
   The edited node must be in a **NotReady** state.
```

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+   .. code-block:: shell
+
+      $ kubectl get nodes
+      node2    NotReady   etcd,master                   19h       v1.11.7
```
You'll probably have more than just `node2` when running `kubectl get nodes`. Either add the _usual_ values in the output, or change the command to `kubectl get node <node_name>`.

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+      $ kubectl get nodes
+      node2    NotReady   etcd,master                   19h       v1.11.7
+
+#. Change the name of the node in the ``yaml`` file used to create it.
```
Maybe add a command to generate a valid YAML manifest from the exisiting `Node` object... Will check how to do that.

gdemonet replied:
Also add a reference to the Quickstart section that details Node manifests.

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+
+      $ kubectl delete node node2
+
+#. Log on the salt master:
```
```suggestion
#. Open a terminal into the :term:`Salt master` container:
```

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+
+   .. code-block:: shell
+
+      $ kubectl -it -n kube-system exec salt-master-bootstrap bash
```
Be careful, static Pod names depend on the actual Node name where they exist.

```suggestion
      $ kubectl -it exec salt-master-<bootstrap_node_name> -n kube-system -c salt-master bash
```

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+
+      $ kubectl -it -n kube-system exec salt-master-bootstrap bash
+
+#. Delete the old node key:
```
```suggestion
#. Delete the now obsolete :term:`Salt minion` key for the changed Node:
```

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+
+      $ salt-key -d <node_key_name>
+
+#. Run the new node name deployment:
```
```suggestion
#. Re-run the deployment for the edited Node:
```

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+          Total states run:     11
+          Total run time:  132.435 s
+
+#. On the bootstrap node, restart the kubelet service:
```
Probably not on Bootstrap:

```suggestion
#. On the edited node, restart the :term:`kubelet` service:
```

gdemonet reviewed (changes requested):
Few changes requested before we can close this PR.

Also, please make sure to:

- capitalize terms such as `Node` and `Bootstrap` when they describe a known entity (K8s Node object, **the** Bootstrap node...)
- introduce terms at least once per document, using the `:term:` inline markup
- make sure to use placeholders consistently across a document (ideally, across the whole docs)

bert-e commented:
# History mismatch



Merge commit #fda28d479d96576ddc4e02579f232c21d423a6c4 on the integration branch
`w/2.1/improvement/1162-doc-change-name-node` is merging a branch which is neither the current
branch `improvement/1162-doc-change-name-node` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/1162-doc-change-name-node` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

ycointe committed:
Add term in the glossary

The documentation provide more term which need to be add to the
glossary.

gdemonet commented on docs/glossary.rst:
```diff
@@ -54,6 +54,12 @@ Glossary
      `kubeconfig <https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/>`_.
 
+   Kubelet
+     The kubelet is the primary “node agent” that runs on each cluster node.
```
Wrong double quotes

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,103 @@
+
+
+#. On the bootstrap node, check the hostname edition incurred a change of
+   status on the bootstrap. The node must be in a **NotReady** status.
```
```suggestion
   status. The edited node must be in a **NotReady** status.
```

ycointe committed:
Minor corrections

Grammatical errors, bad formulations

ycointe committed:
Add reference link to the quickstart guide

Add a link to the quickstart guide documentation
to help user in manifest creation/modification

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,104 @@
+   .. code-block:: shell
+
+      $ kubectl get <node_name>
+      node2    NotReady   etcd,master                   19h       v1.11.7
```
```suggestion
      <node_name>   NotReady   etcd,master                   19h       v1.11.7
```

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,104 @@
+
+   .. code-block:: shell
+
+      $ kubectl apply -f new-node-creation.yaml
```
`new-node-creation.yaml`:

- not a good name (the YAML is a manifest, it describes a Node, not its creation)
- not mentioned before, so could be replaced with `<path to edited manifest>`

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,104 @@
+
+      $ kubectl apply -f new-node-creation.yaml
+
+#. Delete the old node (here ``node2``):
```
```suggestion
#. Delete the old node (here ``<node_name>``):
```

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,104 @@
+
+   .. code-block:: shell
+
+      $ kubectl delete node node2
```
```suggestion
      $ kubectl delete node <node_name>
```

gdemonet commented on docs/quickstart/expansion.rst:
```diff
@@ -113,6 +113,8 @@ The MetalK8s GUI uses :term:`SaltAPI` to orchestrate the deployment.
 Adding a node from the command-line
 -----------------------------------
 
+.. _Creating a manifest:
```
Let's use a standard for references: `<book>-<chapter>-<ref_name>`.
In that case, it would become:

```suggestion
.. _quickstart-expansion-manifest:
```

gdemonet commented on docs/operation/changing_node_hostname.rst:
```diff
@@ -0,0 +1,104 @@
+
+   .. code-block:: shell
+
+      $ salt-key -d <node_key_name>
```
```suggestion
      $ salt-key -d <node_name>
```

gdemonet reviewed (dismissed):
:shipit:

ycointe commented:
@bert-e approve

ycointe commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1162-doc-change-name-node` into
`development/2.0`
* [w/2.1/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1162-doc-change-name-node) will be merged into `development/2.1` 
* [w/2.2/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1162-doc-change-name-node) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

ycointe commented:
@bert-e approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/8bbd80e348072de1e2907fa486610de86623b213) did not succeed in branch w/2.1/improvement/1162-doc-change-name-node.




*The following options are set:* **approve**

ycointe commented:
@bert-e approve

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1162-doc-change-name-node` into
`development/2.0`
* [w/2.1/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1162-doc-change-name-node) will be merged into `development/2.1` 
* [w/2.2/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1162-doc-change-name-node) will be merged into `development/2.2` 
* [w/2.3/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/improvement/1162-doc-change-name-node) will be merged into `development/2.3` 
* [w/2.4/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/improvement/1162-doc-change-name-node) will be merged into `development/2.4` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/ce46d4a5955f0cf0dbce80239d89763e05e18695) did not succeed in branch w/2.2/improvement/1162-doc-change-name-node.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/eb96054cde1b8e479c9aa96591264c02bb7141eb) did not succeed in branch improvement/1162-doc-change-name-node.




*The following options are set:* **approve**

bert-e commented:
# History mismatch



Merge commit #8569a9831e61a2df9677255bdfff496f8ebb60ab on the integration branch
`w/2.1/improvement/1162-doc-change-name-node` is merging a branch which is neither the current
branch `improvement/1162-doc-change-name-node` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/1162-doc-change-name-node` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

ycointe commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1162-doc-change-name-node` into
`development/2.0`
* [w/2.1/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1162-doc-change-name-node) will be merged into `development/2.1` 
* [w/2.2/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1162-doc-change-name-node) will be merged into `development/2.2` 
* [w/2.3/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/improvement/1162-doc-change-name-node) will be merged into `development/2.3` 
* [w/2.4/improvement/1162-doc-change-name-node](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/improvement/1162-doc-change-name-node) will be merged into `development/2.4` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @nootal

* @cmonfort

* @lucieleonard

* @alxf

* @carlito-scality

* @alexandre-allard-scality

* @teddyandrieux

* @slaperche-scality

* @thomasdanan

* @zempashi

* @nicolast

* @sayf-eddine-scality

* @monpote

* @gdemonet

* @tcarmet










*The following options are set:* **approve**

ycointe commented:
@bert-e approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/2a44ae7a2c252280a55a8351155e4fee735c2445) did not succeed in branch improvement/1162-doc-change-name-node.




*The following options are set:* **approve**

bert-e commented:
# History mismatch



Merge commit #eb77eaa7effa7f65f8f5b0f81dfa8a39cfe1c03e on the integration branch
`w/2.1/improvement/1162-doc-change-name-node` is merging a branch which is neither the current
branch `improvement/1162-doc-change-name-node` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/1162-doc-change-name-node` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @nootal

* @cmonfort

* @lucieleonard

* @alxf

* @carlito-scality

* @alexandre-allard-scality

* @teddyandrieux

* @slaperche-scality

* @thomasdanan

* @zempashi

* @nicolast

* @sayf-eddine-scality

* @monpote

* @gdemonet

* @tcarmet










*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/7b57b4ab1d6ee14175509c1bef2dfd16de214f2a) did not succeed in branch improvement/1162-doc-change-name-node.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye ycointe.
